format/File.fbs

```diff
@@ -28,19 +28,11 @@ table Footer {
 
   schema: org.apache.arrow.flatbuf.Schema;
 
-  dictionaries: [ Block ];
-
-  recordBatches: [ Block ];
-}
-
-struct Block {
-
-  offset: long;
-
-  metaDataLength: int;
-
-  bodyLength: long;
+  // Byte offsets into the file for the start of dictionary messages.
+  dictionaries: [ int ];
 
+  // Byte offsets into the file for the start of record batch messages.
+  recordBatches: [ int ];
```

Comment:
One downside of this change getting rid of the Blocks, is that reading a record batch from the file will now require 3 reads from the source (length, metadata, then body) -- before, you could read the whole payload in 1 read if you wanted. not sure how much that matters